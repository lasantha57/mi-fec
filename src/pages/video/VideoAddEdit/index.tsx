import React, { useEffect } from 'react';
import { useLocation, useNavigate } from 'react-router-dom';
import Box from '@mui/material/Box';
import Typography from '@mui/material/Typography';
import Divider from '@mui/material/Divider';
import TextField from '@mui/material/TextField';
import MenuItem from '@mui/material/MenuItem';
import Select, { SelectChangeEvent } from '@mui/material/Select';
import Button from '@mui/material/Button';
import FormControl from '@mui/material/FormControl';
import { Container, InputLabel, OutlinedInput } from '@mui/material';

import { getCategories } from '../../../services/categories';
import { getAuthors } from '../../../services/authors';
import { Category, Author } from '../../../common/interfaces';
import { addOrModifyVideo } from '../../../services/videos';

const DEFAULT_FORMATS = { two: { res: '1080p', size: 1000 } };

export const VideoAddEdit: React.FC = () => {
  const navigate = useNavigate();
  const search = useLocation().search;
  const authorId = new URLSearchParams(search).get('authorId');
  const videoId = new URLSearchParams(search).get('videoId');
  const isEditMode = authorId && videoId;
  const [name, setName] = React.useState('');
  const [categories, setCategories] = React.useState<Category[]>([]);
  const [authors, setAuthors] = React.useState<Author[]>([]);
  const [selectedAuthorId, setSelectedAuthorId] = React.useState<number>(-1);
  const [selectedCategories, setSelectedCategories] = React.useState<number[]>([]);

  const isFormValid = (name !== '' && selectedAuthorId !== -1 && selectedCategories.length > 0) || false;

  useEffect(() => {
    const fetchData = async () => {
      const [categories, authors] = await Promise.all([getCategories(), getAuthors()]);
      setCategories(categories);
      setAuthors(authors);

      if (isEditMode) {
        const auther = authors?.find(author => author.id === parseInt(authorId));
        const video = auther?.videos?.find(video => video.id === parseInt(videoId));

        if (video && auther) {
          setName(video.name);
          setSelectedAuthorId(auther.id);
          setSelectedCategories(video.catIds);
        }
      }
    };
    fetchData();
  }, []);

  const getRandomReleaseDate = () => {
    const maxDate = Date.now();
    const timestamp = Math.floor(Math.random() * maxDate);
    return new Date(timestamp).toISOString().substring(0, 10);
  }

  const onSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();

    const author = authors?.find(author => author.id === selectedAuthorId);
    if (author) {
      if (isEditMode) {
        const video = author?.videos?.find(video => video.id === parseInt(videoId));
        if (video) {
          video.name = name;
          video.catIds = selectedCategories;
        }
      } else {
        const newVideo = {
          id: author.videos.length + 1,
          catIds: selectedCategories,
          name,
          formats: DEFAULT_FORMATS,
          releaseDate: getRandomReleaseDate()
        };
        author.videos = [...author.videos, newVideo];
      }

      try {
        const updatedVideo = await addOrModifyVideo(author);
        if (updatedVideo) {
          redirectToHome();
        }
      } catch (error) {
        console.log(error);
      }
    }
  };

  const redirectToHome = () => {
    navigate('/');
  };

  const handleAuthorChange = (event: SelectChangeEvent<typeof selectedAuthorId>) => {
    const value = event.target.value;
    setSelectedAuthorId(value as typeof selectedAuthorId);
  };

  const handleChangeCategories = (event: SelectChangeEvent<typeof selectedCategories>) => {
    const value = event.target.value;
    setSelectedCategories(typeof value === 'string' ? [] : value);
  };

  return (
    <Container maxWidth={false}>
      <Box>
        <Typography variant="h4" my={3}>
          {isEditMode ? `Edit Video: ${name}` : 'Add Video'}
        </Typography>
        <Divider />
        <Box component="form"
          role={"form"}
          aria-label="Video Information" 
          onSubmit={onSubmit}
          >
          <Box my={3}>
            <TextField
              required
              name="name"
              value={name}
              onChange={(e) => setName(e.currentTarget.value)}
              label="Name"
              aria-label="Video Name"
              sx={{ width: '80%' }}
            />
          </Box>
          <Box my={2}>
            <FormControl sx={{ width: '80%' }}>
              <InputLabel>Author *</InputLabel>
              <Select
                required
                name="author"
                placeholder="Author"
                value={selectedAuthorId}
                onChange={handleAuthorChange}
                input={<OutlinedInput label="Author" />}>
                <MenuItem value="-1" sx={{ display: 'none' }}>
                  Please Select Author
                </MenuItem>
                {
                  authors.map((author) => (
                    <MenuItem key={author.id} value={author.id}>
                      {author.name}
                    </MenuItem>
                  ))
                }
              </Select>
            </FormControl>
          </Box>
          <Box my={3}>
            <FormControl sx={{ width: '80%' }}>
              <InputLabel>Video Categories *</InputLabel>
              <Select
                required
                multiple
                value={selectedCategories}
                onChange={handleChangeCategories}
                input={<OutlinedInput label="Video Categories *" />}>
                {
                  categories.map((category) => (
                    <MenuItem key={category.id} value={category.id}>
                      {category.name}
                    </MenuItem>
                  ))
                }
              </Select>
            </FormControl>
          </Box>
          <Box my={3}>
            <Button
              variant="contained"
              color="primary"
              type="submit"
              disabled={!isFormValid}
              sx={{ mr: 1 }}
            >
              Submit
            </Button>
            <Button
              variant="contained"
              color="secondary"
              onClick={redirectToHome}
            >
              Cancel
            </Button>
          </Box>
        </Box>
      </Box>
    </Container>
  );
};
