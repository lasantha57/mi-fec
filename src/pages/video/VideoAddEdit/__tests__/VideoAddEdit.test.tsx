import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { cleanup, fireEvent, render } from '@testing-library/react';

import VideoAddEdit from '..';

afterEach(cleanup)

describe('Testing VideoAddEdit', () => {
    it('Component should render correctly', () => {
        const tree = renderer
            .create(
                <Router>
                    <VideoAddEdit />
                </Router>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe('Test Edit mode', () => {
        it('With authorId and videoId urls params should render as Edit Video page', async () => {
        
            jest.spyOn(URLSearchParams.prototype, 'get').mockImplementation((key) => key);
    
            const { getByTestId } = render(
                <Router>
                    <VideoAddEdit />
                </Router>
            );
    
            const field = getByTestId('page-title') as HTMLInputElement;
            expect(field).toBeInTheDocument();
            expect(field.textContent).toBe('Edit Video: ');
        });

        it('Without authorId and videoId urls params should render as Add Video page', async () => {
        
            const { getByTestId } = render(
                <Router>
                    <VideoAddEdit />
                </Router>
            );
    
            const field = getByTestId('page-title') as HTMLInputElement;
            expect(field).toBeInTheDocument();
            expect(field.textContent).toBe('Add Video');
        });
    });

    describe('Test state changes', () => {
        it('Inputing text into video name textField should update the state for name', async () => {
        
            jest.spyOn(URLSearchParams.prototype, 'get').mockImplementation((key) => key);
    
            const { getByTestId } = render(
                <Router>
                    <VideoAddEdit />
                </Router>
            );
    
            const field = getByTestId('video-name').querySelector('input') as HTMLInputElement;
            expect(field).toBeInTheDocument();
    
            fireEvent.change(field , { target: { value: 'new video name'}});
            expect(field.value).toBe('new video name');
            
            const pageTitle = getByTestId('page-title') as HTMLInputElement;
            expect(pageTitle.textContent).toBe('Edit Video: new video name');
        });
    })
});