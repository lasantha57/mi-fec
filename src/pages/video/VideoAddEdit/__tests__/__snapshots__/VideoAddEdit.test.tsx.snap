// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing VideoAddEdit Component renders correctly 1`] = `
<div
  className="MuiContainer-root css-ur2jdm-MuiContainer-root"
>
  <div
    className="MuiBox-root css-0"
  >
    <h4
      className="MuiTypography-root MuiTypography-h4 css-19v5luw-MuiTypography-root"
      data-testid="page-title"
    >
      Add Video
    </h4>
    <hr
      className="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
    />
    <form
      aria-label="Video Information"
      className="MuiBox-root css-0"
      onSubmit={[Function]}
      role="form"
    >
      <div
        className="MuiBox-root css-5vb4lz"
      >
        <div
          className="MuiFormControl-root MuiTextField-root css-w3s1g-MuiFormControl-root-MuiTextField-root"
          data-testid="video-name"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor=":r0:"
            id=":r0:-label"
          >
            Name
            <span
              aria-hidden={true}
              className="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id=":r0:"
              name="name"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-1ftyaf0"
              >
                <span>
                  Name
                   
                  *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiBox-root css-1ebprri"
      >
        <div
          className="MuiFormControl-root css-1edjpmh-MuiFormControl-root"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={true}
          >
            Author *
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
            required={true}
            variant="outlined"
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="mui-component-select-author"
              className="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
              id="mui-component-select-author"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              role="button"
              tabIndex={0}
            >
              Please Select Author
            </div>
            <input
              aria-hidden={true}
              className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              disabled={false}
              name="author"
              onAnimationStart={[Function]}
              onChange={[Function]}
              placeholder="Author"
              required={true}
              tabIndex={-1}
              value={-1}
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-1in441m"
              >
                <span>
                  Author
                   
                  *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiBox-root css-5vb4lz"
      >
        <div
          className="MuiFormControl-root css-1edjpmh-MuiFormControl-root"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
          >
            Video Categories *
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
            required={true}
            variant="outlined"
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              className="MuiSelect-select MuiSelect-outlined MuiSelect-multiple MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                className="notranslate"
              >
                ​
              </span>
            </div>
            <input
              aria-hidden={true}
              className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              disabled={false}
              onAnimationStart={[Function]}
              onChange={[Function]}
              required={true}
              tabIndex={-1}
              value=""
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-1ftyaf0"
              >
                <span>
                  Video Categories *
                   
                  *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiBox-root css-5vb4lz"
      >
        <button
          className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root Mui-disabled  css-bh137t-MuiButtonBase-root-MuiButton-root"
          disabled={true}
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="submit"
        >
          Submit
        </button>
        <button
          className="MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-zcbmsk-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          Cancel
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`Testing VideoAddEdit Component should render correctly 1`] = `
<div
  className="MuiContainer-root css-ur2jdm-MuiContainer-root"
>
  <div
    className="MuiBox-root css-0"
  >
    <h4
      className="MuiTypography-root MuiTypography-h4 css-19v5luw-MuiTypography-root"
      data-testid="page-title"
    >
      Add Video
    </h4>
    <hr
      className="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
    />
    <form
      aria-label="Video Information"
      className="MuiBox-root css-0"
      onSubmit={[Function]}
      role="form"
    >
      <div
        className="MuiBox-root css-5vb4lz"
      >
        <div
          className="MuiFormControl-root MuiTextField-root css-w3s1g-MuiFormControl-root-MuiTextField-root"
          data-testid="video-name"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor=":r0:"
            id=":r0:-label"
          >
            Name
            <span
              aria-hidden={true}
              className="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id=":r0:"
              name="name"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-1ftyaf0"
              >
                <span>
                  Name
                   
                  *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiBox-root css-1ebprri"
      >
        <div
          className="MuiFormControl-root css-1edjpmh-MuiFormControl-root"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={true}
          >
            Author *
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
            required={true}
            variant="outlined"
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="mui-component-select-author"
              className="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
              id="mui-component-select-author"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              role="button"
              tabIndex={0}
            >
              Please Select Author
            </div>
            <input
              aria-hidden={true}
              className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              disabled={false}
              name="author"
              onAnimationStart={[Function]}
              onChange={[Function]}
              placeholder="Author"
              required={true}
              tabIndex={-1}
              value={-1}
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-1in441m"
              >
                <span>
                  Author
                   
                  *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiBox-root css-5vb4lz"
      >
        <div
          className="MuiFormControl-root css-1edjpmh-MuiFormControl-root"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
          >
            Video Categories *
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
            required={true}
            variant="outlined"
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              className="MuiSelect-select MuiSelect-outlined MuiSelect-multiple MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                className="notranslate"
              >
                ​
              </span>
            </div>
            <input
              aria-hidden={true}
              className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              disabled={false}
              onAnimationStart={[Function]}
              onChange={[Function]}
              required={true}
              tabIndex={-1}
              value=""
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-1ftyaf0"
              >
                <span>
                  Video Categories *
                   
                  *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiBox-root css-5vb4lz"
      >
        <button
          className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root Mui-disabled  css-bh137t-MuiButtonBase-root-MuiButton-root"
          disabled={true}
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="submit"
        >
          Submit
        </button>
        <button
          className="MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-zcbmsk-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          Cancel
        </button>
      </div>
    </form>
  </div>
</div>
`;
